### PR TITLE
Riley a11y2 & onRouteChange

### PR DESCRIPTION
**Pull request by rileylnapier**
## Description

updates inbox for accessibility

- make sure you can open the inbox with "enter"
- make sure you can tab over all messages, not just anchor links
- make sure you can tab over the "created time"
- make sure you can tab over the actions
- make sure you can pass onRouteChange to the components to implement your own router

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/components/src/index.html [hunk 1]
```diff
@@ -49,15 +49,9 @@
...
-            title: "foo",
-            openLinksInNewTab: false,
-            labels: {
-              closeInbox: "closme",
-              markAllAsRead: "markallread",
-            },
-          },
+        userId: "70f6a4f4-2907-4518-b8f3-b9cfab224764",
+        onRouteChange: (route) => {
+          console.log("route", route);
```
Comment (bblatnick32) [inline packages/components/src/index.html:54]: @rileylnapier Did you mean to have a console log here?

File: packages/react-inbox/src/components/Inbox/index.tsx [hunk 2]
```diff
@@ -139,25 +145,40 @@ const Inbox: React.FunctionComponent<InboxProps> = (props) => {
...
+  const handleIconEvent = useCallback(() => {
+    if (!isOpen) {
+      const now = new Date().getTime();
+      const dateDiff = lastMessagesFetched
+        ? now - lastMessagesFetched
+        : undefined;
+
+      setView("messages");
+
+      if (!dateDiff || dateDiff > 3600000) {
```
Comment (suhasdeshpande) [inline packages/react-inbox/src/components/Inbox/index.tsx:157]: what is 3600000?

File: packages/react-inbox/src/components/Messages2.0/actions/index.tsx [hunk 3]
```diff
@@ -111,28 +118,41 @@ const MessageActions: React.FunctionComponent<{
...
       >
-        {!read && (
-          <MarkRead label={labels?.markAsRead} onClick={handleEvent("read")} />
-        )}
-        {read && (
-          <MarkUnread
-            label={labels?.markAsUnread}
-            onClick={handleEvent("unread")}
+        <TimeAgo
+          tabIndex={0}
```
Comment (suhasdeshpande) [inline packages/react-inbox/src/components/Messages2.0/actions/index.tsx:125]: In general, setting the tabindex of an element to 0 may cause confusion for users who rely on keyboard navigation.

File: packages/react-inbox/src/components/Messages2.0/Message.tsx [hunk 4]
```diff
@@ -194,31 +209,53 @@ const MessageWrapper: React.FunctionComponent<
...
-  } = {};
+    tabIndex: number;
+  } = {
+    tabIndex: 0,
+  };
 
   if (clickAction) {
-    containerProps.href = clickAction;
-    containerProps.onMouseDown = handleClickMessage;
+    if (!courier.onRouteChange) {
```
Comment (bblatnick32) [inline packages/react-inbox/src/components/Messages2.0/Message.tsx:239]: @rileylnapier Might be a little cleaner to do: if (clickAction && !courier.onRouteChange) so as to not nest if statements too many levels

**Review by suhasdeshpande — DISMISSED** [review]
Tiny comment on tabIndex. Considering this is embedded widget, it might be okay but overall, I've some historical difficulties when I introduce tabIndex.

File: packages/react-inbox/src/components/Messages2.0/index.tsx [hunk 5]
```diff
@@ -153,6 +153,7 @@ const Messages: React.ForwardRefExoticComponent<
     },
     ref
   ) => {
+    const [focusedMessageId, setFocusedMessageId] = useState<string>();
```
Comment (bblatnick32) [inline packages/react-inbox/src/components/Messages2.0/index.tsx:156]: useState("") generally safer and won't need the type annotation

File: packages/components/src/index.html (hunk 1 above)
Comment (rileylnapier) [reply packages/components/src/index.html:54]: This is a dev file so it's fine.

File: packages/react-inbox/src/components/Inbox/index.tsx (hunk 2 above)
Comment (rileylnapier) [reply packages/react-inbox/src/components/Inbox/index.tsx:157]: 1 hour

File: packages/react-inbox/src/components/Messages2.0/Message.tsx (hunk 4 above)
Comment (rileylnapier) [reply packages/react-inbox/src/components/Messages2.0/Message.tsx:239]: i would need 2 if statements in that case 

if (clickAction && courier.onRouteChange)

as well as

if (clickAction && !courier.onRouteChange)

so nesting it makes it cleaner in my opinion.  i do agree with you that keeping code all the way to the left is my preference but here id rather keep all the logic related to "clickAction" in one place vs 2 conditionals

File: packages/react-inbox/src/components/Messages2.0/actions/index.tsx (hunk 3 above)
Comment (rileylnapier) [reply packages/react-inbox/src/components/Messages2.0/actions/index.tsx:125]: the point of adding tabIndex is to allow people with keyboard access to access the created time, before they could  not

File: packages/react-inbox/src/components/Messages2.0/index.tsx (hunk 5 above)
Comment (rileylnapier) [reply packages/react-inbox/src/components/Messages2.0/index.tsx:156]: ill do both

**Review by bblatnick32 — APPROVED** [review]

**Pull request merged**